src/angular/11-declarative-state/restaurant.component.spec.ts

```diff
@@ -1,491 +1,466 @@
-    const compiled = fixture.debugElement.nativeElement;
-    let cityOption = compiled.querySelector('select.city option:nth-child(2)');
-    expect(cityOption.textContent).toEqual('Sacramento');
+    const compiled = fixture.nativeElement as HTMLElement;
+    const cityOption = compiled.querySelector(
+      'select[formcontrolname="city"] option:nth-child(2)'
+    ) as HTMLInputElement;
+    expect(cityOption.textContent).toEqual(' Sacramento ');
     expect(cityOption.value).toEqual('Sacramento');
   }));
 
-  it('state dropdown should be disabled until states are populated', <any>fakeAsync((): void => {
-    let stateOutput = null;
-    let stateDisabledOutput = null;
-    const fixture = TestBed.createComponent(RestaurantComponent);
-    const originalGetStates = injectedService.getStates;
-
-    // returns populated data 100ms after subscription
-    injectedService.getStates = () => {
-      return timer(100).pipe(
-          mapTo({
-            data: [
-              {"short": "MO", "name": "Missouri"},
-              {"short": "CA", "name": "California"},
-              {"short": "MI", "name": "Michigan"}]
-          }),
-      );
-    };
+  it('state dropdown should be disabled until states are populated', () => {
+    const stateFormControl = fixture.componentInstance.form.get('state')!;
+    expect(stateFormControl.enabled).toBe(false);
 
     fixture.detectChanges();
-    fixture.componentInstance.states.subscribe((output) => {
-      stateOutput = output;
-    });
-    fixture.componentInstance.stateSelectDisabled.subscribe((output) => {
-      stateDisabledOutput = output;
-    });
 
-    expect(stateDisabledOutput).toBe(true);
+    expect(stateFormControl.enabled).toBe(true);
+  });
 
-    tick(100);
+  it('city dropdown should be disabled until cities are populated', fakeAsync((): void => {
+    fixture.detectChanges(); // detecting changes for createForm func to be called
+    const cityFormControl1 = fixture.componentInstance.form.get('city');
+    expect(cityFormControl1?.enabled).toBe(false);
+    fixture.componentInstance.form.get('state')?.patchValue('CA');
     fixture.detectChanges();
-
-    expect(stateDisabledOutput).toBe(false);
-
-    injectedService.getStates = originalGetStates;
+    const cityFormControl2 = fixture.componentInstance.form.get('city');
+    expect(cityFormControl2?.enabled).toBe(true);
   }));
 
-  it('city dropdown should be disabled until cities are populated', <any>fakeAsync((): void => {
-    const fixture = TestBed.createComponent(RestaurantComponent);
-    let cityDisabledOutput = null;
+  it('should reset list of cities when new state is selected', fakeAsync((): void => {
+    fixture.detectChanges(); // detecting changes for createForm func to be called
 
-    fixture.detectChanges(); //detecting changes for createForm func to be called
-    fixture.componentInstance.citySelectDisabled.subscribe((output) => {
-      cityDisabledOutput = output;
+    let restaurantOutput: any;
```

Comment:
Can we not type this?

Reply:
@m-thompson-code now typed. had to use the `!` otherwise it errors saying typescript throws `is used before being assigned`